infrastructure/network/connmanager/outgoing_connections.go

```diff
@@ -24,46 +28,26 @@ func (c *ConnectionManager) checkOutgoingConnections(connSet connectionSet) {
 
 	connectionsNeededCount := c.targetOutgoing - len(c.activeOutgoing)
 	connectionAttempts := connectionsNeededCount * 2
-	for i := 0; i < connectionAttempts; i++ {
+	netAddresses := c.addressManager.RandomAddresses(connectionAttempts, connectedAddresses)
```

Comment:
When you get to call RandomAddresses, you have to make sure that connectedAddresses also includes the incoming addresses.